module/synchronization/metrics.go

```diff
@@ -0,0 +1,138 @@
+package synchronization
+
+import (
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type SynchronizationMetrics interface {
+	// record pruned blocks. requested and received times might be zero values
+	PrunedBlockById(status *Status)
+
+	PrunedBlockByHeight(status *Status)
```

Comment:
My understanding of why this metrics collector can't live in the `module/metrics` package is because it is depending on `Status`, which is defined in `module/synchronization`, creating the circular dependency. 

I suggest that we move the `Status` model to a separate model package so that we can maintain a consistent organization of the metric collectors. For example:
* Create a package `model/chainsync` (for context, @smnzhu has a pending [PR](https://github.com/onflow/flow-go/pull/2292) which changes the name of this module to `chainsync` -- we can preemptively name the model package accordingly)
* Add `Range`, `Batch`, `Status` to `model/chainsync`
* Then, we can move `SynchronizationMetrics` (can also preemptively rename this to `ChainSyncMetrics`) to `module/metrics.go and the implementation to the `module/metrics` package